streams/src/test/java/org/apache/kafka/streams/TopologyTest.java

```diff
@@ -274,18 +273,7 @@ public void shouldThrowOnUnassignedStateStoreAccess() throws Exception {
                 goodNodeName)
             .addProcessor(badNodeName, new LocalMockProcessorSupplier(), sourceNodeName);
 
-        try {
-            new ProcessorTopologyTestDriver(streamsConfig, topology.internalTopologyBuilder);
-        } catch (final StreamsException e) {
-            final Throwable cause = e.getCause();
-            if (cause != null
-                && cause instanceof TopologyBuilderException
-                && cause.getMessage().equals("Invalid topology building: Processor " + badNodeName + " has no access to StateStore " + LocalMockProcessorSupplier.STORE_NAME)) {
-                throw (TopologyBuilderException) cause;
-            } else {
-                throw new RuntimeException("Did expect different exception. Did catch:", e);
-            }
-        }
+        new ProcessorTopologyTestDriver(streamsConfig, topology.internalTopologyBuilder);
```

Comment:
We should keep try-catch and check the error message. Also add a `fail` if not exception is thrown.
```
try {
    new ProcessorTopologyTestDriver(streamsConfig, topology.internalTopologyBuilder);
    fail("Should have thrown StreamsException");
} catch (final StreamException e) {
...
```